assets/app/actions/siteActions.js

```diff
@@ -55,6 +55,23 @@ export default {
     }).catch(error => alertActions.httpError(error.message));
   },
 
+  uploadFile(site, path, fileData, message) {
```

Comment:
I included the file decoding in the actual component. I left a comment in there about moving it to an action creator, wanted to get some feedback on if that seemed like a good idea. I believe this confirms that it would be!